File: algorithms/medium/sequential_digits.ex
defmodule Solution do
  @spec sequential_digits(low :: integer, high :: integer) :: [integer]
  def sequential_digits(low, high) do
    1..9
    |> Enum.reduce([], &aux(&1, &2, low, high))
    |> Enum.sort()
  end

  def aux(n, acc, low, high) do
    d = rem(n, 10)

    cond do
      n < low && d != 0 -> aux(n * 10 + d + 1, acc, low, high)
      n <= high && d != 0 -> aux(n * 10 + d + 1, [n | acc], low, high)
      true -> acc
    end
  end
end

File: algorithms/medium/sequential_digits.rs
impl Solution {
    pub fn sequential_digits(low: i32, high: i32) -> Vec<i32> {
        let mut ans = Vec::new();
        for i in 1..=9 {
            let mut n = i;
            while n < low && n % 10 != 0 {
                n = n * 10 + n % 10 + 1;
            }
            while n <= high && n % 10 != 0 {
                ans.push(n);
                n = n * 10 + n % 10 + 1;
            }
        }
        ans.sort();
        ans
    }
}
